api/src/main/java/org/xwiki/filemanager/internal/reference/DefaultUniqueDocumentReferenceGenerator.java

```diff
@@ -92,7 +99,17 @@ public void initialize() throws InitializationException
     {
         // Initialize the cache.
         CacheConfiguration cacheConfiguration = new CacheConfiguration();
-        cacheConfiguration.setConfigurationId(UUID.randomUUID().toString());
+        String uniqueDocumentReference = "unique.documentReference";
+        String namespace = null;
+        // Change name of configuration depending on where is the application installed, on farm or subwiki.
+        if (componentManager instanceof NamespacedComponentManager) {
+            namespace = ((NamespacedComponentManager) componentManager).getNamespace();
+        }
+        if (namespace != null) {
+            namespace = namespace.split("wiki:", 2)[1];
```

Comment:
This will throw an ``ArrayIndexOutOfBounds`` if the namespace doesn't contain "wiki:" (it should most of the time, but better be safe). You could use [``XarHandlerUtils#getWikiFromNamespace``](https://github.com/xwiki/xwiki-platform/blob/master/xwiki-platform-core/xwiki-platform-extension/xwiki-platform-extension-handlers/xwiki-platform-extension-handler-xar/src/main/java/org/xwiki/extension/xar/internal/handler/XarHandlerUtils.java#L42) (it's internal API, but it's OK for us).